From 5dbddc3d4ac81d707b347e74f2700f58e5644e4f Mon Sep 17 00:00:00 2001
From: Harshitha Prem <quic_hprem@quicinc.com>
Date: Mon, 6 Mar 2023 20:57:17 +0530
Subject: [PATCH] cfg80211: precac check for self-managed wiphy

Currently, precac regulatory domain is inferred based
on the wiphy regulatory domain and cfg80211 regulatory
database.

Some of the countries has some mismatch between these two
domains for self-managed driver, e.g., LK.

Hence, add changes to determine precac regulatory domain
based on the wiphy regulatory settings if it is self-managed
wiphy.

Signed-off-by: Harshitha Prem <quic_hprem@quicinc.com>
---
 net/wireless/reg.c | 7 +++++++
 1 file changed, 7 insertions(+)

--- a/net/wireless/reg.c
+++ b/net/wireless/reg.c
@@ -4254,7 +4254,8 @@ bool regulatory_pre_cac_allowed(struct w
 		return pre_cac_allowed;
 	}
 
-	if (regd->dfs_region == wiphy_regd->dfs_region &&
+	if ((regd->dfs_region == wiphy_regd->dfs_region ||
+	    wiphy->regulatory_flags & REGULATORY_WIPHY_SELF_MANAGED) &&
 	    wiphy_regd->dfs_region == NL80211_DFS_ETSI)
 		pre_cac_allowed = true;
 
